flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdWindowProperties.scala

```diff
@@ -252,13 +254,40 @@ class FlinkRelMdWindowProperties private extends MetadataHandler[FlinkMetadata.W
   }
 
   def getWindowProperties(
-      rel: StreamPhysicalWindowAggregate,
+      rel: StreamPhysicalWindowAggregateBase,
       mq: RelMetadataQuery): RelWindowProperties = {
+    rel match {
+      case _: StreamPhysicalWindowAggregate =>
+        val aggregate = rel.asInstanceOf[StreamPhysicalWindowAggregate]
+        getWindowAggregateWindowProperties(
+          aggregate.grouping.length + aggregate.aggCalls.size(),
+          aggregate.namedWindowProperties,
+          aggregate.windowing.getWindow,
+          aggregate.windowing.getTimeAttributeType
+        )
+      case _: StreamPhysicalGlobalWindowAggregate =>
+        val aggregate = rel.asInstanceOf[StreamPhysicalGlobalWindowAggregate]
+        getWindowAggregateWindowProperties(
+          aggregate.grouping.length + aggregate.aggCalls.size(),
+          aggregate.namedWindowProperties,
+          aggregate.windowing.getWindow,
+          aggregate.windowing.getTimeAttributeType
+        )
+      case _ =>
```

Comment:
Can the input of window join be `StreamPhysicalGroupWindowAggregateBase`? If the answer is yes, it may need to add the case of `StreamPhysicalGroupWindowAggregateBase`. Or the same thing will happen if the inputs of window join are `StreamPhysicalGroupWindowAggregateBase`.

Reply:
`StreamPhysicalGroupWindowAggregateBase` is a base class for `StreamPhysicalGroupWindowAggregate` & `StreamPhysicalGlobalGroupWindowAggregate`. 
The input of window join might be  `StreamPhysicalGroupWindowAggregate` or `StreamPhysicalGlobalGroupWindowAggregate`

Reply:
> Can the input of window join be `StreamPhysicalGroupWindowAggregateBase`? If the answer is yes, it may need to add the case of `StreamPhysicalGroupWindowAggregateBase`. Or the same thing will happen if the inputs of window join are `StreamPhysicalGroupWindowAggregateBase`.

According to the doc[1], I think the input of window join cannot be a legacy window group aggregate. The join on top of the group window aggregate will be translated to the regular join, other than window join.
> Windowing TVFs is a replacement of legacy [Grouped Window Functions](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/window-agg/#group-window-aggregation-deprecated). Windowing TVFs is more SQL standard compliant and more powerful to support complex window-based computations, e.g. Window TopN, Window Join. However, [Grouped Window Functions](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/window-agg/#group-window-aggregation) can only support Window Aggregation.

[1] https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/queries/window-tvf/

Reply:
Thank you very much for your answer.